app/models/user.rb

```diff
@@ -107,20 +107,32 @@ def gpo_verification_pending_profile?
   end
 
   def pending_profile
-    pending = profiles.where(deactivation_reason: :in_person_verification_pending).or(
-      profiles.where.not(gpo_verification_pending_at: nil),
-    ).or(
-      profiles.where.not(fraud_review_pending_at: nil),
-    ).or(
-      profiles.where.not(fraud_rejection_at: nil),
-    ).order(created_at: :desc).first
-
-    return unless pending.present?
-    return if pending.password_reset?
-    return if pending.encryption_error? || pending.verification_cancelled?
-    return if active_profile.present? && active_profile.activated_at > pending.created_at
-
-    pending
+    return @pending_profile if defined?(@pending_profile)
+
+    @pending_profile = begin
```

Comment:
Is there a concern about it becoming stale? I'm not sure if there's a place where we call `pending_profile`, update it, and then call the method again, but it could cause issues if we do at some point.

Reply:
Yes, there is a concern. We do call it this way in the tests. We do not currently do that in the code that runs the application